templates/common/infra/bicep/core/security/role.bicep

```diff
@@ -0,0 +1,20 @@
+param principalId string
+
+@allowed([
+  'Device'
+  'ForeignGroup'
+  'Group'
+  'ServicePrincipal'
+  'User'
+])
+param principalType string = 'ServicePrincipal'
+param roleDefinitionId string
+
+resource role 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid(subscription().id, resourceGroup().id, principalId, roleDefinitionId)
+  properties: {
+    principalId: principalId
+    principalType: principalType
+    roleDefinitionId: resourceId('Microsoft.Authorization/roleDefinitions', roleDefinitionId)
+  }
+}
```

Comment:
Since there isn't any target specified - I would assume this is for subscription wide role assignments?

Reply:
It inherits the scope it is deployed in, so it will have scope of the rg by default.  Let's add scope if needed in the future.